pkg/provisioner/reconciler/claim-reconciler/reconiler.go

```diff
@@ -226,41 +248,84 @@ func (r *objectBucketClaimReconciler) createConfigMap(options *api.BucketOptions
 	if err != nil {
 		return nil, fmt.Errorf("error composing configmap for ObjectBucketClaim %s/%s: %v", options.ObjectBucketClaim.Namespace, options.ObjectBucketClaim.Name, err)
 	}
+
 	r.logD.Info("creating Configmap", "namespace", configMap.Namespace, "name", configMap.Name)
-	if err = util.CreateUntilDefaultTimeout(configMap, r.client, r.retryInterval, r.retryTimeout); err != nil {
+	err = util.CreateUntilDefaultTimeout(configMap, r.client, r.retryInterval, r.retryTimeout)
+	if err != nil {
 		return nil, fmt.Errorf("unable to create ConfigMap %q for claim %v: %v", configMap.Name, options.ObjectBucketClaim.Name, err)
 	}
 	return configMap, nil
 }
 
+func (r *objectBucketClaimReconciler) createUntilDefaultTimeout(obj runtime.Object) error {
+	return wait.PollImmediate(r.retryInterval, r.retryTimeout, func() (done bool, err error) {
+		err = r.client.Create(r.ctx, obj)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return false, err
+		}
+		return true, nil
+	})
+}
+
 // shouldProvision is a simplistic check on whether this obc is a concern for this provisioner.
 // Down the road, this will perform a broader set of checks.
 func (r *objectBucketClaimReconciler) shouldProvision(obc *v1alpha1.ObjectBucketClaim) bool {
-	if obc == nil {
-		r.logI.Info("nil OBC, assuming delete event")
-		return false
-	}
 
-	class, err := util.StorageClassForClaim(obc, r.client, r.ctx)
-	if err != nil {
-		klog.Errorf("cannot provision: %v", err)
+	if obc.Spec.ObjectBucketName != "" {
+		r.logI.Info("provisioning already completed", "ObjectBucket", obc.Spec.ObjectBucketName)
 		return false
 	}
-	if class.Provisioner != r.provisionerName {
-		r.logI.Info("claim provisioner does not match expected provisioner", "claim provisioner", class.Provisioner, "should match", r.provisionerName)
+	if obc.Spec.StorageClassName == "" {
```

Comment:
This doesn't handle the case of obc.spec.storageClass specified but the actual SC doesn't exist? This mayb e okay if this case is logged and retried.